packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -150,6 +150,14 @@ export const render = async (req, res, next) => {
         })
     }
 
+        // Serialize the application extensions that have an async getRoutes where the key is the name of the extension
+    // and the value is the serialized extension data.
+    const serializedExtensions = Object.fromEntries(
+        applicationExtensions
+            .filter((extension) => extension.isRoutesAsync)
+            .map((extension) => [extension.getName(), extension.serialize()])
+    )
```

Comment:
only serialize the routes for extensions that have implemented an `async getRoutes`